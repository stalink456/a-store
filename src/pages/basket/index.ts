export { Basket as default } from './basket';
