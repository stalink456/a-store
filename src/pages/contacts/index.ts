export { Contacts as default } from './contacts';
