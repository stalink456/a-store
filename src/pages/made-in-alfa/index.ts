export { MadeInAlfa as default } from "./made-in-alfa";
