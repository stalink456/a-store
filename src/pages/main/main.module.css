@import url("../../css/variables.css");

.main {
  display: flex;
  flex-wrap: nowrap;

  width: calc(100vw - 20px);
}

@media (width <= 960px) {
  .main {
    flex-wrap: wrap;

    width: 100vw;
  }
}
