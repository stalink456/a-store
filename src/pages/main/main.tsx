import React from 'react';
import { MainPicture } from 'components/ui-components/main-picture';
import madeInAlfa from 'images/MadeInAlfa.jpeg';
import yourDesign from 'images/YourDesign.jpeg';

import styles from './main.module.css';

export const Main: React.FC = () => {
  return (
    <div className={styles.main}>
      <MainPicture
        src={madeInAlfa}
        text='Сделано в Альфе'
        route='/made-in-alfa'
      />
      <MainPicture src={yourDesign} text='Свой дизайн' route='/your-design' />
    </div>
  );
};
