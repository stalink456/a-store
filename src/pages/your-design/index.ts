export { YourDesign as default } from "./your-design";
