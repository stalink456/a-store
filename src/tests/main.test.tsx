import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Main } from 'pages/main';

describe('Main component', () => {
  it('should render Main component', () => {
    const { baseElement } = render(<Main />, { wrapper: MemoryRouter });

    expect(baseElement).toMatchSnapshot();
  });

  it('should render Main component with text', async () => {
    render(<Main />, { wrapper: MemoryRouter });

    await screen.findByText('Свой дизайн');
  });

  it("shouldn't render Main component with non-existent text", async () => {
    render(<Main />, { wrapper: MemoryRouter });

    expect(screen.queryByText('non-existent text')).not.toBeInTheDocument();
  });
});
