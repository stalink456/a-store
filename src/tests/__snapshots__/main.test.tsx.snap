// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render Main component 1`] = `
<body>
  <div>
    <div
      class="main"
    >
      <div
        class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as main__picture"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <a
            href="/made-in-alfa"
          >
            <h1
              class="typography__component_19zgr typography__styrene-medium_pxbut typography__bold_pxbut"
            >
              Сделано в Альфе
            </h1>
            <div
              class="skeleton__component_1bshy skeleton__animate_1bshy"
            >
              <img
                alt="Сделано в Альфе"
                class=""
                src=""
              />
            </div>
          </a>
        </div>
      </div>
      <div
        class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as main__picture"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <a
            href="/your-design"
          >
            <h1
              class="typography__component_19zgr typography__styrene-medium_pxbut typography__bold_pxbut"
            >
              Свой дизайн
            </h1>
            <div
              class="skeleton__component_1bshy skeleton__animate_1bshy"
            >
              <img
                alt="Свой дизайн"
                class=""
                src=""
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
